### ISSUE TITLE
auto-scrolling isolated

### ISSUE THREAD

**Issue by figoyouwei** [figoyouwei/taipy_success#28]
Hey, 

Sure, I have isolated the issue NOT using LangChain and update the code. Just plugin your openai keys to the .env, it shall work.

It reproduces the issue.
<img width="946" alt="Screenshot 2024-10-21 at 11 12 02" src="https://github.com/user-attachments/assets/7cf7f07f-f082-4b2d-808e-b8139d8252a5">

**Comment by AlexandreSajus** [comment]
Where have you reproduced the issue exactly? We are looking for a simple piece of code (less than 100 lines) that still has the issue, so the issue is isolated, and I can fix it or send it to R&D.

**Comment by figoyouwei** [comment]
The whole chatbot thing is under:
https://github.com/figoyouwei/taipy_success/tree/main/chatbot

1. Go into this directory, add .env with openai keys as below:
```
OPENAI_API_KEY=
OPENAI_MODEL=
```
2. If necessary
```
pip install requirements.txt
```
3. Run main.py and it shall reproduce the issue:)

**Comment by AlexandreSajus** [comment]
I found the bug. It is caused by setting the `height` property in the chat control: https://github.com/Avaiga/taipy/issues/2111

R&D will solve this in another release.

In the meantime, you can fix your issue by not setting the `height` property on `tgb.chat` in  `pages/chat.py` line 71:
```python
with tgb.Page() as page_chat:
    # Doc for chat control: https://docs.taipy.io/en/develop/manuals/userman/gui/viselements/generic/chat/
    tgb.chat(
        # Note: messages is actually the "var_name" in the evaluate function
        messages="{messages}",
        users=users,
        on_action=evaluate,
        sender_id="Human",
        # NOTE: to solve icon issue
        show_sender=True,
    )
```

**Comment by figoyouwei** [comment]
ok, I see. So when another release is out, I could set the height again, is that correct? Can I expect this release by the end of this month, October?

**Comment by AlexandreSajus** [comment]
Yes, probably November

**Comment by figoyouwei** [comment]
ok, let's see early November and keep this issue open.

**Comment by AlexandreSajus** [comment]
This is fixed in taipy 4.0.1

**Comment by figoyouwei** [comment]
I have upgraded to v4.0.1, but ... it doesn't seem to get fixed, do you need me to send you a video?

**Comment by AlexandreSajus** [comment]
Make sure you scroll all the way down to trigger auto-scroll. Can you send me a video? Which piece of code are you running? Please check that you are using taipy-enterprise 4.0.1 using `pip list`

I don't trigger the issue with this example code:
```python
from taipy.gui import Gui
import taipy.gui.builder as tgb

messages = []


def evaluate(state, var_name: str, payload: dict):
    state.messages.append(
        [10, "The MP5 is pretty fun to shoot actually, virtually no recoil", "Robot"]
    )
    state.messages = state.messages


with tgb.Page() as page_home:
    tgb.chat(
        messages="{messages}",
        on_action=evaluate,
        height="80vh",
    )

gui = Gui(page=page_home)
gui.run()
```

**Comment by figoyouwei** [comment]
Does that work with normally taipy v4.0.1, not enterprise v4.0.1?

**Comment by AlexandreSajus** [comment]
Yes, when there too many messages on the screen, scroll down to the last message. If you add a new message, the chat will scroll down automatically, even in 4.0.1

**Comment by figoyouwei** [comment]
Yeah, I send an video

**Comment by figoyouwei** [comment]
And now when this bug could be fixed with new release? Is that possible next week?

**Comment by AlexandreSajus** [comment]
This will be fixed in a new release for December. I don't think it will be out by next week

**Comment by figoyouwei** [comment]
And could you push this thing a bit before Christmas? I really need this to be fixed ... it has been a month...

**Comment by FlorianJacta** [comment]
This issue has been resolved (see [here](https://github.com/Avaiga/taipy/issues/2281#event-15630237208)). We are preparing a 4.0.2 version so that you have this fixed as soon as possible

**Comment by figoyouwei** [comment]
Thank, please release the version before your Christmas holiday if possible:)

**Comment by AlexandreSajus** [comment]
Hey Youwei! We just released a version that fixes your issue. We are currently testing it to ensure it does not bring new bugs. You can already test it by pip installing the .whl file contained in this attachment:
[taipy-4.0.2.dev2-py3-none-any.zip](https://github.com/user-attachments/files/18227723/taipy-4.0.2.dev2-py3-none-any.zip)

**Comment by figoyouwei** [comment]
Hey,

Nice to hear from you guys during the Christmas week. I think I will pass the .whl dev phrase, I have faith on you and please let me know when there is an official release that I can pip.

**Comment by figoyouwei** [comment]
I installed 4.0.2 and didn't see any improvement... Please watch the video below, thanks.

https://github.com/user-attachments/assets/83629a6a-9bfe-40f1-8dca-eaa1194ff665

**Comment by AlexandreSajus** [comment]
I found an issue in the code: in `chatbot/pages/chat.py`, remove line 92:
```python
    # NOTE: update frontend
    state.partial_chat.update_content(state, page_chat)
```
You don't need to update the partial and updating the chat partial resets the scroll state. This fixes the issue on my end:
![image](https://github.com/user-attachments/assets/9d551b01-dc07-4e56-b650-7184d6c1fd38)


Can you also make sure both taipy and taipy-enterprise are on 4.0.2? Running `pip list` should display:
```bash
taipy                                    4.0.2
taipy-common                             4.0.2
taipy-core                               4.0.2
taipy-enterprise                         4.0.2
taipy-gui                                4.0.2
taipy-rest                               4.0.2
taipy-templates                          4.0.2
```
You can run `pip install taipy==4.0.2` after installing taipy enterprise to ensure taipy goes to 4.0.2
